util/metrics/prometheus/prometheus.go

```diff
@@ -148,3 +157,26 @@ func (m *counterMetric) Values() map[string]float64 {
 	}
 	return values
 }
+
+// Clock sets the metric's value to the given duration in seconds
+func (m *gaugeMetric) Clock(t time.Duration, fields ...string) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	m.met.WithLabelValues(fields...).Set(t.Seconds())
```

Comment:
Can we have `gaugeMetric` track the units desired and scale accordingly, rather than continuing to force everything into seconds?

Reply:
Sure, I suppose. Prometheus doesn't seem to have a way to assign "metadata" in this way, other than changing the description.

I could change this interface so that NewDuration takes a unit (seconds, milliseconds), and then includes that unit with the description so we avoid a unitless number, but it didn't seem helpful to support `200 milliseconds` instead of `.2 seconds` for individual controllers.

Reply:
I want to support 200 milliseconds

Reply:
Given that we can spend anywhere from 10s to 20m updating a group, seconds seems like an appropriate unit (why I changed it from milliseconds).

OTOH when measuring latency that averages 50ms, reporting this as 50ms rather than 0.005ms tends to be easier to visualize and reason about.

Reply:
Hmm, maybe that was just this https://github.com/GoogleCloudPlatform/testgrid/compare/fa39445790be66e9d611f32dced19728fd673ff9..fa39445790be66e9d611f32dced19728fd673ff9~1